### PR TITLE
[Tiny PR] Media Placeholder: Allow SVG as URI

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Ok, ok, SVGs in WordPress is a controversial topic. They cannot be uploaded. And the way towards that _seems_ long as there are a bunch of security issues with SVGs. See https://core.trac.wordpress.org/ticket/24251.

But what if we don't upload them? What if we just encode them and use them in a data URI? This is safe as any scripts that could be embedded in an SVG will not be executed. The SVG is not embedded in the page.

Wait? Aren't data URIs too big to store in the post content? Yes, base64 encoded text would be long, but this isn't necessary. We can just encode the text.

So imagine I upload something like:

```html
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"> 
<circle cx="50" cy="18.4" r="18.4" fill="#f00" /> 
<script> 
 alert( 'Hello there!' ); 
</script> 
</svg>
```

The result would be:

```html
<img src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'%3E%3Ccircle cx='50' cy='18.4' r='18.4' fill='%23f00' /%3E%3Cscript%3E alert( 'Hello there!' ); %3C/script%3E%3C/svg%3E" alt="" width="196" height="196"/>
```

## How has this been tested?

Add an image block and click "Upload", then select and SVG.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/block-editor/src/components/media-placeholder/index.js [hunk 1]
```diff
@@ -130,8 +131,23 @@ export class MediaPlaceholder extends Component {
 				setMedia = onSelect;
 			}
 		} else {
+			const [ file ] = files;
+
+			if ( file.type === 'image/svg+xml' ) {
+				file.text().then( ( source ) => {
+					this.props.onSelectURL(
```
Comment (jorgefilipecosta) [inline packages/block-editor/src/components/media-placeholder/index.js]: Should we check if onSelectURL is available?

File: packages/block-editor/src/components/media-placeholder/index.js [hunk 2]
```diff
@@ -130,8 +131,23 @@ export class MediaPlaceholder extends Component {
 				setMedia = onSelect;
 			}
 		} else {
+			const [ file ] = files;
+
+			if ( file.type === 'image/svg+xml' ) {
```
Comment (jorgefilipecosta) [inline packages/block-editor/src/components/media-placeholder/index.js]: Should we limit this logic to files smaller than a given amount of bytes?

File: packages/block-editor/src/components/media-placeholder/index.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-editor/src/components/media-placeholder/index.js]: Oh, sure, I didn't know it's optional. :)

File: packages/block-editor/src/components/media-placeholder/index.js (hunk 2 above)
Comment (ellatrix) [reply packages/block-editor/src/components/media-placeholder/index.js]: That sounds like a good idea. Like me check some browser limitations for data URIs.

**Comment by gwwar** [comment]
Interesting. This might work? I'm not sure how cautious we'll be on the security side of things. Note that this should only be set on the image tag. In other contexts <iframe>, <object> or <embed> can execute code. For example https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#You_can_EMBED_SVG_which_can_contain_your_XSS_vector

**Comment by draganescu** [comment]
I love this idea!

**Comment by paaljoachim** [comment]
It looks like this PR has gone stale.
It could use a status update.
Thanks!

**Review by jorgefilipecosta — COMMENTED** [review]
There is one more check that I think we should do. Users without the ability to output unsafe HTML e.g: authors can not include code with an SVG so I guess in these cases our mechanism should do nothing.

**Comment by kathrynwp** [comment]
@jorgefilipecosta Hey there, just circling back to see whether this is likely to be picked up and merged, or whether it should be closed out at this stage. Let me know, thanks!

**Comment by jorgefilipecosta** [comment]
> @jorgefilipecosta Hey there, just circling back to see whether this is likely to be picked up and merged, or whether it should be closed out at this stage. Let me know, thanks!

I think this PR is useful and could be merged. I guess we should just add a check for the capability to not have this functionality on users without permission to use unsafe HTML.

**Comment by priethor** [comment]
@ellatrix, is this PR still relevant? If so, maybe @jorgefilipecosta can wrap it once he comes back from AFK.

**Comment by ellatrix** [comment]
I'm not going to work on it, anyone should feel free to take it and start a new PR.

**Pull request closed without merging**